src/integration/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/EhrRequestTest.java

```diff
@@ -1,204 +1,221 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.ConversationTransferStatus.INBOUND_COMPLETE;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.ConversationTransferStatus.INBOUND_CONTINUE_REQUEST_SENT;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.ConversationTransferStatus.INBOUND_REQUEST_SENT;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.ConversationTransferStatus.INBOUND_TIMEOUT;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.Layer.CONVERSATION;
+import static uk.nhs.prm.repo.ehrtransferservice.database.enumeration.TransferTableAttribute.DESTINATION_GP;
+import static uk.nhs.prm.repo.ehrtransferservice.utils.MessageParsingUtility.getMessageId;
+import static uk.nhs.prm.repo.ehrtransferservice.utils.TestDataLoaderUtility.getTestDataAsString;
+
+@SpringBootTest
+@ActiveProfiles("test")
+@WireMockTest(httpPort = 8080)
+@ExtendWith(SpringExtension.class)
+@ExtendWith(ForceXercesParserExtension.class)
+@ContextConfiguration(classes = LocalStackAwsConfig.class)
+class EhrRequestTest {
+    private static final String NHS_NUMBER = "9798548754";
+    private static final UUID INBOUND_CONVERSATION_ID = UUID.fromString("ce3aad10-9b7c-4a9b-ab87-a9d6521d61b2");
+    private static final String NEMS_MESSAGE_ID = "eefe01f7-33aa-45ed-8aac-4e0cf68670fd";
+    private static final String NEMS_EVENT_LAST_UPDATED = "2017-11-01T15:00:33+00:00";
+    private static final String SOURCE_GP = "B14758";
+    private static final String PRESIGNED_URL = "http://localhost:8080/presigned/url";
+    private static final String REPO_INCOMING_MESSAGE = getRepoIncomingMessage();
+
+    @Autowired
+    private TransferService transferService;
+
+    @Autowired
+    private TransferTrackerDbUtility transferTrackerDbUtility;
+
+    @Autowired
+    private SqsQueueUtility sqsQueueUtility;
+
+    @Value("${aws.repoIncomingQueueName}")
+    private String repoIncomingQueueName;
+
+    @Value("${activemq.inboundQueue}")
+    private String inboundQueue;
+
+    @Value("${gp2gpMessengerAuthKey}")
+    private String gp2gpMessengerAuthKey;
+
+    @Value("${ehrRepoAuthKey}")
+    private String ehrRepoAuthKey;
+
+    @AfterEach
+    void afterEach() {
+        sqsQueueUtility.purgeQueue(repoIncomingQueueName);
+        transferService.updateConversationTransferStatus(INBOUND_CONVERSATION_ID, INBOUND_COMPLETE);
+        transferTrackerDbUtility.deleteItem(INBOUND_CONVERSATION_ID, CONVERSATION);
```

Comment:
- Move this logic into the test.
- New assertion to ensure in-memory map **doesn't** contain `INBOUND_CONVERSATION_ID`.